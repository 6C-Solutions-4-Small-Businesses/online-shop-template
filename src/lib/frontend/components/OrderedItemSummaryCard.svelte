<script lang="ts">
  import Image from '$lib/frontend/components/Image.svelte'

  export let id: string
  export let name: string
  export let image: string
  export let selectedQuantity: number
  export let price: number
</script>

<div data-testid="{id}" class="h-32 xl:h-36 flex bg-white rounded-md shadow-md">
  <Image imageRemoteUrl="{image}" name={name} classes="w-2/5 p-2 h-full product-image object-contain"/>
  <div class="w-3/5 flex flex-col justify-between hover:bg-orange-100 pl-2">
    <div class="flex justify-between">
      <div class="font-thin flex justify-center items-center">{`${selectedQuantity} x ${name}`}</div>
    </div>
    <div class="flex justify-between h-10">
      <div class="w-4/12 font-thin flex items-center">${(price / 100).toFixed(2)}</div>
    </div>
  </div>
</div>

<style lang="css">
  .product-image {
    aspect-ratio: 1;
  }
</style>
