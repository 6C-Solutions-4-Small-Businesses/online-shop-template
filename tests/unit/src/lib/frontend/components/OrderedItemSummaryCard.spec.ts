import {render} from '@testing-library/svelte'
import {describe, expect} from 'vitest'
import OrderedItemSummaryCard from '$lib/frontend/components/OrderedItemSummaryCard.svelte'

describe('OrderedItemSummaryCard', () => {
    it('should have match snapshot', () => {
        const {container} = render(OrderedItemSummaryCard)

        expect(container).toMatchSnapshot()
    })
})
