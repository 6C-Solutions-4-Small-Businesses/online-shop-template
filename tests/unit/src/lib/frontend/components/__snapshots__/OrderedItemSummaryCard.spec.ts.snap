// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OrderedItemSummaryCard > should have match snapshot 1`] = `
<body>
  <div>
    <div
      class="h-32 xl:h-36 flex bg-white rounded-md shadow-md s-ReROHsK2djRp"
    >
      <img
        class="w-2/5 p-2 h-full product-image object-contain"
        width="160"
      />
      <!--&lt;Image&gt;-->
       
      <div
        class="w-3/5 flex flex-col justify-between hover:bg-orange-100 pl-2 s-ReROHsK2djRp"
      >
        <div
          class="flex justify-between s-ReROHsK2djRp"
        >
          <div
            class="font-thin flex justify-center items-center s-ReROHsK2djRp"
          >
            undefined x undefined
          </div>
        </div>
         
        <div
          class="flex justify-between h-10 s-ReROHsK2djRp"
        >
          <div
            class="w-4/12 font-thin flex items-center s-ReROHsK2djRp"
          >
            $
            NaN
          </div>
        </div>
      </div>
    </div>
    <!--&lt;OrderedItemSummaryCard&gt;-->
  </div>
</body>
`;
